### PR TITLE
Add experimental extension to access firmware security version (#211)

### PR DESCRIPTION
**Pull request by sanilkumar0**
Resolves: oneapi-src/level-zero-spec#211

### CODE REVIEW

File: scripts/sysman/EXT_Exp_FirmwareSecurityVersion.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    s=tags['$s']
+    S=s.upper()
+%>
+:orphan:
+
+.. _ZES_extension_ras_state:
```
Comment (aravindksg) [inline scripts/sysman/EXT_Exp_FirmwareSecurityVersion.rst:11]: extension name to be changed

File: scripts/sysman/EXT_Exp_FirmwareSecurityVersion.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+======================================
+ Firmware security firmware version
```
Comment (aravindksg) [inline scripts/sysman/EXT_Exp_FirmwareSecurityVersion.rst:14]: firmware word is repeated

File: scripts/sysman/EXT_Exp_FirmwareSecurityVersion.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+* Functions
+
+    * ${s}FirmwareGetSecurityVersionExp
+    * ${s}FirmwareSetSecurityVersionExp
+
+* Enumerations
+
+    * ${s}_firmware_security_exp_version_t
+
+Security Version
```
Comment (aravindksg) [inline scripts/sysman/EXT_Exp_FirmwareSecurityVersion.rst:29]: could you add more literature on objectives for this extension and programming example

File: scripts/sysman/frmwareSecurityVersion.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#
+# Copyright (C) 2023 Intel Corporation
```
Comment (aravindksg) [inline scripts/sysman/frmwareSecurityVersion.yml:2]: copyright year

File: scripts/sysman/frmwareSecurityVersion.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+desc: "Anti rollback security version Extension Version(s)"
+version: "1.9"
+name: $s_firmware_security_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get anti roll-back security firmware version number of the current running firmware"
```
Comment (aravindksg) [inline scripts/sysman/frmwareSecurityVersion.yml:29]: IIRC, the IGSC team mentioned the number here is the version number that would get commited when the user calls "SetSecurityVersion" which is the trigger to write the number right?

also, do we need another out param or another API to provide a "min allowed version number"

File: scripts/sysman/frmwareSecurityVersion.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+name: SetSecurityVersionExp
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_firmware_handle_t
+      name: hFirmware
+      desc: "[in] Handle for the component."
+    - type: char
+      name: "version[$S_STRING_PROPERTY_SIZE]"
```
Comment (aravindksg) [inline scripts/sysman/frmwareSecurityVersion.yml:57]: setSecurityVersion IIUC really is just a trigger right? I don't think we even need an input version number here?

**Comment by aravindksg** [comment]
@joshuaranjan @saik-intel @AshwinKumarKulkarni

**Review by joshuaranjan — COMMENTED** [review]
No additional points

File: scripts/sysman/frmwareSecurityVersion.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get anti roll-back security firmware version number of the current running firmware"
+version: "1.9"
+class: $sFirmware
+name: GetSecurityVersionExp
+details:
+    - "The application may call this function from simultaneous threads."
```
Comment (saik-intel) [inline scripts/sysman/frmwareSecurityVersion.yml:34]: Do we really need Exp API ? As the API is new and its not conflicting or breaking old API.

File: scripts/sysman/frmwareSecurityVersion.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Set anti roll-back security firmware version number"
+version: "1.9"
+class: $sFirmware
+name: SetSecurityVersionExp
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
```
Comment (saik-intel) [inline scripts/sysman/frmwareSecurityVersion.yml:52]: Are we sure that this function implementation should be lock free?  while setting SVN via IGSC, we might need lock right? @aravindksg  @joshuaranjan  @AshwinKumarKulkarni   @bellekal

File: scripts/sysman/frmwareSecurityVersion.yml (hunk 7 above)
Comment (aravindksg) [reply scripts/sysman/frmwareSecurityVersion.yml:34]: prefer experimental APIs for now, it has better flexibility. we can always promote them in future as needed once APIs are stable

File: scripts/sysman/frmwareSecurityVersion.yml (hunk 8 above)
Comment (bellekal) [reply scripts/sysman/frmwareSecurityVersion.yml:52]: I think it says **should** and not **must**. From https://spec.oneapi.io/level-zero/latest/core/INTRO.html#terminology **should** means it is recommended and not mandated. So i think it is ok to use "should be lock free".

**Review by wdamon-intel — CHANGES_REQUESTED** [review]
@sanilkumar0 please add narrative/example to the extension programming guide

File: scripts/sysman/frmwareSecurityVersion.yml (hunk 5 above)
Comment (sanilkumar0) [reply scripts/sysman/frmwareSecurityVersion.yml]: GetSecurityVersionExp returns the 'min version allowed', which comes from the fuses

File: scripts/sysman/frmwareSecurityVersion.yml (hunk 6 above)
Comment (sanilkumar0) [reply scripts/sysman/frmwareSecurityVersion.yml:57]: Good catch.. Yes, we don't need an input here.

**Review by wdamon-intel — APPROVED** [review]

**Pull request merged**